programs/locker/src/instructions/v3/claim.rs

```diff
@@ -0,0 +1,224 @@
+use crate::util::{transfer_to_user_v3, MemoTransferContext};
+use crate::*;
+use anchor_spl::memo::Memo;
+use anchor_spl::token_interface::{Mint, TokenAccount, TokenInterface};
+use safe_math::SafeMath;
+use solana_program::hash::hashv;
+use util::{
+    close, parse_remaining_accounts, AccountsType, ParsedRemainingAccounts,
+    TRANSFER_MEMO_CLAIM_VESTING,
+};
+
+const LEAF_PREFIX: &[u8] = &[0];
+const INTERMEDIATE_PREFIX: &[u8] = &[1];
+
+#[derive(AnchorSerialize, AnchorDeserialize)]
+/// Accounts for [locker::create_vesting_escrow].
+pub struct ClaimV3Params {
+    pub vesting_start_time: u64,
+    pub cliff_time: u64,
+    pub frequency: u64,
+    pub cliff_unlock_amount: u64,
+    pub amount_per_period: u64,
+    pub number_of_period: u64,
+    pub max_amount: u64,
+    pub proof: Vec<[u8; 32]>,
+}
+
+impl ClaimV3Params {
+    pub fn get_total_locked_amount(&self) -> Result<u64> {
+        let total_locked_amount = self
+            .cliff_unlock_amount
+            .safe_add(self.amount_per_period.safe_mul(self.number_of_period)?)?;
+        Ok(total_locked_amount)
+    }
+
+    pub fn verify_recipient(&self, recipient: Pubkey, root: [u8; 32]) -> Result<()> {
+        let total_locked_amount = self.get_total_locked_amount()?;
+        let node = hashv(&[
```

Comment:
Node should be build from `cliff_unlock_amount `, `amount_per_period ` and `number_of_period `, not by `total_locked_amount`

Reply:
`total_locked_amount` derive from (`cliff_unlock_amount`, `amount_per_period`, `number_of_period`). 
But each time recipient claim we will pass it from client
that's why we should use `total_locked_amount` to verify when recipient claim.

Reply:
i fixed it.

Reply:
at the first time, becasue `total_locked_amount` derived from `amount_per_period`, `cliff_unlock_amount`, `number_of_period`, make constraint to ensure that all params is consistency and recipient can not fake value.

Reply:
total_locked_amount could be the same, but they can use different values for number_of_period, amount_per_period and cliff_unlock_amount

Reply:
okay, i got it.

Reply:
why not add `vesting_start_time `? if we dont push all params in merkle tree, recipient can be easily to fake that